### PR TITLE
Feature/gdev 123 suggestions

### PR DESCRIPTION
**Pull request by KerstenBreuer**
Here are some suggestions, please let me know if you like to discuss them

### CODE REVIEW

File: example-config.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
-host: 127.0.0.1
-port: 8080
-log_level: info
-workers: 2
-auto_reload: False
+smtp_server: smtp.gmail.com
+smtp_port: 587
+smtp_username: your-username-here
+smtp_password: your-password-here
```
Comment (ac-jorellanaf) [inline example-config.yaml]: This file is public. Why would you store credentials here?

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -26,9 +26,6 @@ services:
     # Use "forwardPorts" in **devcontainer.json** to forward an app port locally.
     # (Adding the "ports" property to this file will not forward from a Codespace.)
 
-    networks:
```
Comment (ac-jorellanaf) [inline .devcontainer/docker-compose.yml]: Already outdated. Done in 131.

File: example-config.yaml (hunk 1 above)
Comment (KerstenBreuer) [reply example-config.yaml]: It's an example file that just lists all config options for a user. It's not used anywhere directly. Please consider it as part of the documentation

Comment (KerstenBreuer) [reply example-config.yaml]: Please have a look here: https://github.com/ghga-de/microservice-repository-template/tree/dev#configuration
We should at some point also migrate that Readme to this repo.

Comment (ac-jorellanaf) [reply example-config.yaml:4]: Do environment variables overwrite what is in this file?

Comment (ac-jorellanaf) [reply example-config.yaml]: Ah, I didn't realise it was the example yaml. Ok. But even then wouldn't it be preferable if the user weren't enticed to put their credentials in a yaml to try to follow this example?

File: example-config.yaml [hunk 3]
```diff
@@ -1,5 +1,7 @@
-host: 127.0.0.1
-port: 8080
-log_level: info
-workers: 2
-auto_reload: False
+smtp_server: smtp.gmail.com
```
Comment (ac-jorellanaf) [inline example-config.yaml:1]: Can we make up a random smtp server instead?

File: sandbox_notification/config.py [hunk 4]
```diff
@@ -17,20 +17,20 @@
...
     """Config parameters and their defaults."""
 
-    # config parameter needed
-    # are inherited from BaseSettings
-
-    smtpserv: str = Field(..., env="sandbox-notification_smtpserv")
-    smtpport: int = Field(..., env="sandbox-notification_smtpport")
-    smtpusername: str = Field(..., env="sandbox-notification_smtpusername")
-    smtppassword: str = Field(..., env="sandbox-notification_smtppassword")
+    smtp_server: str
```
Comment (ac-jorellanaf) [inline sandbox_notification/config.py:27]: Does this still let settings be loaded from env?

File: sandbox_notification/core/send.py [hunk 5]
```diff
@@ -28,29 +28,29 @@ class MaxAttemptsReached(Exception):
...
 def send_email(data: dict):
     """
     Sending email
     """
 
-    attempt = 1
-    while attempt <= MAX_ATTEMPTS:
+    config = get_config()
+
+    for attempt in range(0, config.max_attempts):
```
Comment (ac-jorellanaf) [inline sandbox_notification/core/send.py:38]: Either change this to `range(1, config.max_attempts + 1)`
or change the printout to `{attempt + 1}/{config.max_attempts}` (or even ` {attempt}/{config.max_attempts - 1}`).

**Comment by ac-jorellanaf** [comment]
@KerstenBreuer Please resolve the conflicts.

File: example-config.yaml (hunk 1 above)
Comment (KerstenBreuer) [reply example-config.yaml]: It's up to the user to choose what fit's him best. But we could add a warning.

File: example-config.yaml (hunk 3 above)
Comment (KerstenBreuer) [reply example-config.yaml:1]: It's an example, so why don't we put an example smtp here? Most people have a gmail account this example might help them deploy this even faster

Comment (ac-jorellanaf) [reply example-config.yaml:1]: Sure.

File: example-config.yaml (hunk 1 above)
Comment (ac-jorellanaf) [reply example-config.yaml]: That would be good I think.

Comment (KerstenBreuer) [reply example-config.yaml]: I pushed it.

File: sandbox_notification/config.py (hunk 4 above)
Comment (KerstenBreuer) [reply sandbox_notification/config.py:27]: Yes definitely, that's the whole point of our ghga_service_chassis_lib

File: sandbox_notification/core/send.py (hunk 5 above)
Comment (KerstenBreuer) [reply sandbox_notification/core/send.py:38]: good point

**Comment by KerstenBreuer** [comment]
> @KerstenBreuer Please resolve the conflicts.

✅

**Review by ac-jorellanaf — APPROVED** [review]

**Pull request merged**